### PR TITLE
MM-15699 Add E2E tests for search date filter

### PR DESCRIPTION
**Pull request by thekiiingbob**
#### Summary
Add tests based on the search date filter tab of the RC Testing spread sheet. All the tests cases there should be covered, albeit reorganized and "translated" for cypress.

#### Ticket Link
[JIRA](https://mattermost.atlassian.net/browse/MM-15699)

### CODE REVIEW

File: cypress/support/ui_commands.js [hunk 1]
```diff
@@ -5,6 +5,19 @@
 
 import * as TIMEOUTS from '../fixtures/timeouts';
 
+Cypress.Commands.overwrite('get', (originalFn, selector, options) => {
+    function isCssSelector(string) {
+        const regex = /[a-z]*?(@|#|\.).+/;
```
Comment (cometkim) [inline cypress/support/ui_commands.js]: This is way not enough to parse CSS selectors. nestings(`.parent > .child`), combinations(`[attr=val]`), etc... Maybe we need to use 3rd party css parser library.

I suggest to define new command `getByTestId()` just like react-testing-library does.

File: cypress/support/api_commands.js [hunk 2]
```diff
@@ -258,29 +258,30 @@ Cypress.Commands.add('loginAsNewUser', (user = {}) => {
...
+            teamsToAdd = teams;
+        } else {
+            // Get teams, select the first three, and add new user to that team
+            cy.request('GET', '/api/v4/teams').then((teamsResponse) => {
+                // Verify we have at least 2 teams in the response to add the user to
+                expect(teamsResponse).to.have.property('body').to.have.length.greaterThan(1);
+
+                // Pull out only the first 2 teams
+                const [team1, team2] = teamsResponse.body;
+                teamsToAdd = [team1, team2].map((t) => t.id);
```
Comment (cometkim) [inline cypress/support/api_commands.js]:
```suggestion
                teamsToAdd = teamsResponse.body.slice(0, 2).map((t) => t.id);
```

shortcut?

File: cypress/support/api_commands.js [hunk 3]
```diff
@@ -169,10 +169,10 @@ Cypress.Commands.add('apiGetTeams', () => {
  * This API assume that the user is logged in and has cookie to access
  * @param {Array} preference - a list of user's preferences
  */
-Cypress.Commands.add('apiSaveUserPreference', (preferences = []) => {
+Cypress.Commands.add('apiSaveUserPreference', (preferences = [], userId = 'me') => {
```
Comment (cometkim) [inline cypress/support/api_commands.js]: :+1:

File: cypress/support/ui_commands.js (hunk 1 above)
Comment (thekiiingbob) [reply cypress/support/ui_commands.js]: I'll make it an alternate command for now. We may want to look at just using cypress-testing-library to get `getByTestId()` for free. I'm thinking I should just update that `data-` attribute to just be `data-testid` so we are somewhat future proofed then.

Another approach I was thinking, and I'd love your input, was to preface our testId selectors with our own symbol, like `cy.get('@my-test-id')`. On one side of the coin, I like the simplicity of having just on command ,`get`, to handle getting elements. But on the other we may lose clarity and add a certain level of "magic".

Comment (cometkim) [reply cypress/support/ui_commands.js]: I think conventions that are not properly guided are literally "magic". In that sense, I usually (include this case) prefer the API approach.

Comment (cometkim) [reply cypress/support/ui_commands.js]: And also using `.get('@...')` would be too easy to conflict with regular aliases.

Comment (thekiiingbob) [reply cypress/support/ui_commands.js]: `@` was a bad example then :)

Separate command it is!

**Comment by thekiiingbob** [comment]
@cometkim & @saturninoabril latest pushed up. Ended up adding cypress-testing-library, I like the options it adds for us.

File: package.json [hunk 4]
```diff
@@ -75,6 +75,7 @@
     "@babel/preset-env": "7.4.5",
     "@babel/preset-react": "7.0.0",
     "@babel/runtime": "7.4.5",
+    "@testing-library/cypress": "4.0.3",
```
Comment (saturninoabril) [inline package.json]: 👍

**Pull request merged**